astropy/io/fits/tests/test_header.py

```diff
@@ -99,148 +83,6 @@ def test_add_commentary(self):
-
-        hdu = fits.ImageHDU()
-        hdu.header.update('MYKEY', 'FOO', 'BAR')
-        hdu.header.fromTxtFile(self.temp('header.txt'))
-        # hdu.header should have MYKEY keyword, and also adds PCOUNT and
-        # GCOUNT, giving it 3 more keywords in total than the original
-        assert len(hdul[0].header.ascard) == len(hdu.header.ascard) - 3
-        assert 'MYKEY' in hdu.header
-        assert 'EXTENSION' not in hdu.header
-        assert 'SIMPLE' in hdu.header
-
-        with ignore_warnings():
-            hdu.writeto(self.temp('test.fits'), output_verify='ignore',
-                        clobber=True)
-        hdul2 = fits.open(self.temp('test.fits'))
-        assert len(hdul2) == 2
-        assert 'MYKEY' in hdul2[1].header
-
-    @ignore_warnings(AstropyDeprecationWarning)
-    def test_update_comment(self):
-        hdul = fits.open(self.data('arange.fits'))
-        hdul[0].header.update('FOO', 'BAR', 'BAZ')
-        assert hdul[0].header['FOO'] == 'BAR'
-        assert hdul[0].header.ascard['FOO'].comment == 'BAZ'
-
-        hdul.writeto(self.temp('test.fits'))
-
-        hdul = fits.open(self.temp('test.fits'), mode='update')
-        hdul[0].header.ascard['FOO'].comment = 'QUX'
-        hdul.close()
-
-        hdul = fits.open(self.temp('test.fits'))
-        assert hdul[0].header.ascard['FOO'].comment == 'QUX'
-
-    @ignore_warnings(AstropyDeprecationWarning)
-    def test_long_commentary_card(self):
-        # Another version of this test using new API methods is found in
-        # TestHeaderFunctions
-        header = fits.Header()
-        header.update('FOO', 'BAR')
-        header.update('BAZ', 'QUX')
-        longval = 'ABC' * 30
-        header.add_history(longval)
-        header.update('FRED', 'BARNEY')
-        header.add_history(longval)
-
-        assert len(header.ascard) == 7
-        assert header.ascard[2].key == 'FRED'
-        assert str(header.cards[3]) == 'HISTORY ' + longval[:72]
-        assert str(header.cards[4]).rstrip() == 'HISTORY ' + longval[72:]
-
-        header.add_history(longval, after='FOO')
-        assert len(header.ascard) == 9
-        assert str(header.cards[1]) == 'HISTORY ' + longval[:72]
-        assert str(header.cards[2]).rstrip() == 'HISTORY ' + longval[72:]
-
-    @ignore_warnings(AstropyDeprecationWarning)
-    def test_wildcard_slice(self):
-        """Test selecting a subsection of a header via wildcard matching."""
-
-        header = fits.Header()
-        header.update('ABC', 0)
-        header.update('DEF', 1)
-        header.update('ABD', 2)
-        cards = header.ascard['AB*']
-        assert len(cards) == 2
-        assert cards[0].value == 0
-        assert cards[1].value == 2
-
-    @ignore_warnings(AstropyDeprecationWarning)
-    def test_assign_boolean(self):
-        """
-        Regression test for https://aeon.stsci.edu/ssb/trac/pyfits/ticket/123
-
-        Tests assigning Python and Numpy boolean values to keyword values.
-        """
```

Comment:
I'm hesitant about outright removing this test and some of the ones above.  Especially if they're regression tests.  Instead they should be rewritten to just not use the deprecated APIs.  I have a vague memory of keeping some tests using deprecated APIs just as a way to exercise those APIs.  But it doesn't mean the test itself isn't still relevant.

At the very least I'd want to know what coverage is changed by removing these tests.